append_lifecycle.go

```diff
@@ -258,11 +258,29 @@ func NewAppender(ctx context.Context, d Driver, opts *AppendOptions) (*Appender,
 		ctx, span := tracer.Start(ctx, "tessera.Appender.Add")
 		defer span.End()
 
-		return t.Add(ctx, entry)
+		// NOTE: We memoize the returned value here so that repeated calls to the returned
```

Comment:
Indentation looks a bit off.
```
		// NOTE: We memoize the returned value here so that repeated calls to the returned
		//       future don't result in unexpected side-effects from inner AddFn functions
		//       being called multiple times.
		//		 Currently this is the outermost wrapping of Add so we do the memoization
		//		 here, if this changes, ensure that move the memoization call so that this
		//       remains true.
```